sos/collector/clusters/juju.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2023 Canonical Ltd., Chi Wai Chan <chiwai.chan@canonical.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import json
+import collections
+from sos.collector.clusters import Cluster
+
+
+def _parse_option_string(strings):
+    """Parse commad separated string."""
+    if not strings:
+        return []
+    return [string.strip() for string in strings.split(",")]
+
+
+class juju(Cluster):
+    """
+    The juju cluster profile is intended to be used on juju managed clouds.
+    It"s assumed that `juju` is installed on the machine where `sos` is called,
+    and that the juju user has superuser privilege to the current controller.
+
+    By default, the sos reports will be collected from all the applications in
+    the current model. If necessary, you can filter the nodes by models /
+    applications / units / machines with cluster options.
+
+    Example:
+
+    sos collect --cluster-type juju -c "juju.models=sos" -c "juju.apps=a,b,c"
```

Comment:
Should we change something like `sos collect --cluster-type juju -c "juju.models=lma juju.apps=graylog" -c "juju.models=openstack juju.apps=glance"` to give the ability to collect from multiple models at the same time? Or is it overkill?

Reply:
we should not